src/repositories/updater.rs

```diff
@@ -0,0 +1,366 @@
+use crate::error::Result;
+use crate::repositories::{GitHub, GitLab};
+use crate::utils::MetadataPackage;
+use crate::{db::Pool, Config};
+use chrono::{DateTime, Utc};
+use log::{debug, info, trace, warn};
+use once_cell::sync::Lazy;
+use postgres::Client;
+use regex::Regex;
+use std::collections::{HashMap, HashSet};
+use std::fmt;
+
+pub trait RepositoryForge {
+    /// Result used both as the `host` column in the DB and to match repository URLs during
+    /// backfill.
+    fn host(&self) -> &str;
```

Comment:
This *must* be a &'static str; we send it to the database and a dynamic string could cause SQL injections. That doesn't happen now because we hard-code all the updaters, but making it &'static ensures it doesn't happen in the future either.
```suggestion
    fn host(&self) -> &'static str;
```

Reply:
:+1:

Reply:
Until `Box::leak(String::into_boxed_str(String::new("foobar")))`.

Reply:
And suddenly we entered a new dimension. :p

Reply:
> Until Box::leak(String::into_boxed_str(String::new("foobar"))).

Yes, but someone will notice that in review and ask why you're leaking a string. Just returning a `format!` string isn't as obviously wrong.